arches/app/models/graph.py

```diff
@@ -2828,99 +2593,103 @@ def restore_state_from_serialized_graph(self, serialized_graph):
         Restores a Graph's state from a serialized graph, and creates a
         new editable_future_graph
         """
-        models.NodeGroup.objects.filter(
-            pk__in=[
-                nodegroup.pk
-                for nodegroup in self.get_nodegroups(force_recalculation=True)
-            ]
-        ).delete()
-        models.Node.objects.filter(
-            pk__in=[node.pk for node in self.nodes.values()]
-        ).delete()
-        models.Edge.objects.filter(
-            pk__in=[edge.pk for edge in self.edges.values()]
-        ).delete()
-        models.CardModel.objects.filter(
-            pk__in=[card.pk for card in self.cards.values()]
-        ).delete()
-        models.CardXNodeXWidget.objects.filter(
-            pk__in=[
-                card_x_node_x_widget.pk
-                for card_x_node_x_widget in self.widgets.values()
-            ]
-        ).delete()
+        with transaction.atomic():
+            editable_future_graph = Graph.objects.filter(
+                source_identifier_id=self.pk
+            ).first()
+            if editable_future_graph:
+                editable_future_graph.delete()
```

Comment:
If you want to one-shot this, since we're not using the model instance.

Given that this is the Graph proxy, it probably does shave off all the `__init__()` queries.
```suggestion
            Graph.objects.filter(source_identifier_id=self.pk).delete()
```

Reply:
ah -- this is this way because this calls the graph proxy model's `delete` method, while `Graph.objects.filter(source_identifier_id=self.pk).delete()` would skip it.

Reply:
Alright, got it. It would be nice if these were vanilla model methods, but that's not a matter for today.